lib/fluent/plugin/out_splunk_hec.rb

```diff
@@ -51,12 +53,13 @@ class SplunkHECOutput < ObjectBufferedOutput
 
     def configure(conf)
       super
-      raise ConfigError, "'channel' parameter is required when 'use_ack' is true" if @use_ack && !@channel
+      raise ConfigError, "'channel' parameter is required when 'use_ack' is true" if @use_ack && !@auto_generate_channel && !@channel
       raise ConfigError, "'ack_interval' parameter must be a non negative integer" if @use_ack && @ack_interval < 0
       raise ConfigError, "'event_key' parameter is required when 'raw' is true" if @raw && !@event_key
-      raise ConfigError, "'channel' parameter is required when 'raw' is true" if @raw && !@channel
+      raise ConfigError, "'channel' parameter is required when 'raw' is true" if @raw && !@auto_generate_channel && !@channel
       
       @default_sourcetype = @sourcetype if @sourcetype && !@default_sourcetype
+      @mychannel = @auto_generate_channel ? SecureRandom.uuid : @channel
 
```

Comment:
This is complicated.
How about putting following line after `super`?

```
@channel = SecureRandom.uuid if @auto_generate_channel
```

Reply:
This is the 

> Can you follow the DCO? https://github.com/fluent/fluent-plugin-splunk/pull/26/checks?check_run_id=482405499

I cannot... its signing with my official mail id.. which I don't to want to expose